src/properties/property-effects.html

```diff
@@ -1131,23 +1346,32 @@
        * 
        * If the root of the path is a managed property, returns a normalized
        * string path suitable for setting into the system via `_setProperty`/
-       * `_setPendingProperty`.
+       * `_setPendingProperty`.  A null path is returned if the path being set
+       * fails a dirty check, unless `fromPath` was true (in which case no dirty
+       * check is performed since this is a notification of change to a shared
+       * path).
        *
        * `path` can be a path string or array of path parts as accepted by the
        * public API.
        *
        * @param {string} path Path to set
        * @param {*} value Value to set
+       * @param {boolean} fromPath If the value being set was from a path; in
+       *   this case the value was shared, so no dirty check is performed.
        * @return {?string} If the root property was managed, the normalized
        *   string representation of the path, otherwise returns falsey.
        * @protected
        */
-      _setPathOrUnmanagedProperty(path, value) {
+      _setPathOrUnmanagedProperty(path, value, fromPath) {
```

Comment:
Changed to perform a dirty check of a path against the actual object it refers to, and skip setting & return null if the value hasn't changed.  This puts the dirty check for path changes at the entry point (this is only called from public entry points), allowing per-instance caching of paths to be eliminated.

The `fromPath` argument indicates this method is being called as a result of a path notification, meaning it's a set to a shared object and has hence the dirty check should be skipped as it was already dirty checked upon entry.